File: compact_db.js

const fs = require('fs');
const path = 'd:/auction-multiplayer/script.js';

try {
    const data = fs.readFileSync(path, 'utf8');
    const lines = data.split(/\r?\n/);
    let newLines = [];
    let inDb = false;
    let buffer = "";

    const playerStartRegex = /^\s*"[^"]+":\s*\{/;

    for (let i = 0; i < lines.length; i++) {
        const line = lines[i];
        
        if (line.includes("const PLAYER_DATABASE = {")) {
            inDb = true;
            newLines.push(line);
            continue;
        }

        if (inDb && line.trim() === "};") {
            if (buffer) {
                newLines.push(buffer);
                buffer = "";
            }
            inDb = false;
            newLines.push(line);
            continue;
        }

        if (inDb) {
            const stripped = line.trim();
            
            if (stripped.startsWith("//") || stripped === "") {
                if (buffer) {
                    newLines.push(buffer);
                    buffer = "";
                }
                newLines.push(line);
                continue;
            }

            if (playerStartRegex.test(stripped)) {
                if (buffer) {
                    newLines.push(buffer);
                }
                buffer = line.trimEnd();
            } else {
                if (buffer) {
                    buffer += " " + stripped;
                } else {
                   // Should technically not happen if structure matches expectation, 
                   // but strictly keeping non-matching lines safe
                   newLines.push(line);
                }
            }

            if (buffer && (buffer.endsWith("},") || buffer.endsWith("}"))) {
                newLines.push(buffer);
                buffer = "";
            }

        } else {
            newLines.push(line);
        }
    }

    fs.writeFileSync(path, newLines.join('\n'), 'utf8');
    console.log("Success");

} catch (err) {
    console.error(err);
}

File: compact_db.py

import re

file_path = 'd:/auction-multiplayer/script.js'

with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

new_lines = []
in_db = False
buffer = ""
db_indent = ""

# Regex to identify the start of a player entry
# It looks like: "Name": {
player_start_pat = re.compile(r'^\s*"[^"]+":\s*\{')

for i, line in enumerate(lines):
    # Detect start of PLAYER_DATABASE
    if "const PLAYER_DATABASE = {" in line:
        in_db = True
        new_lines.append(line)
        continue
    
    # Detect end of PLAYER_DATABASE
    if in_db and line.strip() == "};":
        if buffer:
            # flush any remaining buffer (shouldn't really happen if structure is consistent)
            new_lines.append(buffer)
            buffer = ""
        in_db = False
        new_lines.append(line)
        continue

    if in_db:
        stripped = line.strip()
        # If it's a comment or empty line, keep it as is, unless we are buffering a player
        if stripped.startswith("//") or not stripped:
            if buffer:
                new_lines.append(buffer + "\n")
                buffer = ""
            new_lines.append(line)
            continue
            
        # If we encounter a new player start, flush previous buffer
        if player_start_pat.match(stripped):
            if buffer:
                new_lines.append(buffer + "\n")
            # Start new buffer
            # Keep indentation of the key
            # But we want to collapse the rest
            # The structure is: "Name": {\n ... \n  },
            buffer = line.rstrip() # Start the line
        else:
            # It's part of the current object content
            # Add to buffer, removing extra whitespace
            if buffer:
                buffer += " " + stripped
            else:
                # Should not happen if format is consistent, but if so just add
                new_lines.append(line)
                
        # If line ends with "}," or "}" it might be the end of the object
        # but we handle that by accumulating until the next start or end of DB?
        # Actually, "}," indicates end of player object usually.
        # Let's check if the buffer ends with "}," or "}"
        if buffer and (buffer.endswith("},") or buffer.endswith("}")):
             # We found the end of an entry
             # Clean up the buffer: remove multiple spaces, fix colons syntax consistency if needed
             # The user code has `key: value,`
             
             new_lines.append(buffer + "\n")
             buffer = ""
             
    else:
        new_lines.append(line)

# Write back
with open(file_path, 'w', encoding='utf-8') as f:
    f.writelines(new_lines)
